From 946bf3b2a4aa51760074b7578be3a228970ba53f Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Wed, 2 Aug 2017 09:45:41 +0800
Subject: [PATCH 2274/5242] MLK-16108 PCI: imx: turn on pd for imx8mq pcie

commit  d2cf1e1db1ce07f6b99bad6c622bc8216c572e59 from
https://source.codeaurora.org/external/imx/linux-imx.git

Root cause:
Poewr domain of the PCIEs are turned off, and
not turned on properly in previous ATF.

The PDs of PCIE1/2 have the dependency.
Both of the PDs should be operated at same time.
This issue is gone after update the PDs operations
in ATF.
In order to make sure that the PDs are turned on,
Turn power domain for imx8mq pcie explicitly in
driver.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/pci/controller/dwc/pci-imx6.c |   12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/drivers/pci/controller/dwc/pci-imx6.c b/drivers/pci/controller/dwc/pci-imx6.c
index 0dc57fd..850adbd 100644
--- a/drivers/pci/controller/dwc/pci-imx6.c
+++ b/drivers/pci/controller/dwc/pci-imx6.c
@@ -1293,6 +1293,7 @@ static int imx6_pcie_establish_link(struct imx6_pcie *imx6_pcie)
 			clk_disable_unprepare(imx6_pcie->pcie_inbound_axi);
 		release_bus_freq(BUS_FREQ_HIGH);
 		if ((imx6_pcie->variant == IMX7D)
+				|| (imx6_pcie->variant == IMX8MQ)
 				|| (imx6_pcie->variant == IMX8QM)
 				|| (imx6_pcie->variant == IMX8QXP))
 			pm_runtime_put_sync(pci->dev);
@@ -1314,8 +1315,10 @@ static int imx6_pcie_host_init(struct pcie_port *pp)
 	struct imx6_pcie *imx6_pcie = to_imx6_pcie(pci);
 
 	/* enable disp_mix power domain */
-	if ((imx6_pcie->variant == IMX7D) || (imx6_pcie->variant == IMX8QM)
-				|| (imx6_pcie->variant == IMX8QXP))
+	if ((imx6_pcie->variant == IMX7D)
+			|| (imx6_pcie->variant == IMX8MQ)
+			|| (imx6_pcie->variant == IMX8QM)
+			|| (imx6_pcie->variant == IMX8QXP))
 		pm_runtime_get_sync(pci->dev);
 
 	imx6_pcie_assert_core_reset(imx6_pcie);
@@ -2045,7 +2048,10 @@ static int __init imx6_pcie_probe(struct platform_device *pdev)
 		pp->ops = &imx6_pcie_host_ops;
 
 		/* enable disp_mix power domain */
-		if (imx6_pcie->variant == IMX7D)
+		if ((imx6_pcie->variant == IMX7D)
+				|| (imx6_pcie->variant == IMX8MQ)
+				|| (imx6_pcie->variant == IMX8QM)
+				|| (imx6_pcie->variant == IMX8QXP))
 			pm_runtime_get_sync(dev);
 
 		imx6_pcie_assert_core_reset(imx6_pcie);
-- 
1.7.9.5

